proposals/js-string-builtins/Overview.md

```diff
@@ -163,6 +163,20 @@ JS Strings are semantically a sequence of 16-bit code units (referred to as char
 
 There is however, the Encoding API for `TextEncoder`/`TextDecoder` which can be used for UTF-8 support. However, this is technically a separate spec from JS and may not be available on all JS engines (in practice it's available widely). This proposal exposes UTF-8 data conversions using this API under separate `wasm:text-encoder` `wasm:text-decoder` interfaces which are available when the host implements these interfaces.
 
+## String constants
+
+String constants may be defined in JS and made available to wasm through a variety of means.
+
+The simplest way is to have a module import each string as an immutable global. This can work for small amounts of strings, but has a high cost for when the number of string constants is very large.
+
+This proposal adds an extension to JS-API compile routine to support optimized 'imported string constants' to address this use-case.
+
+The `WebAssemblyCompileOptions` is extended with a `boolean importedStringConstants` flag. When this is set, the module may define imports of the form `(import "'" %stringConstant (global externref))`, and the JS-API will use the provided `%stringConstant%` import field name to be the value of the global. This allows for any UTF-8 string to be imported with minimal overhead.
```

Comment:
nit: missing `%` after `%stringConstant` in the `(import...` example. You may also want to wrap that in double quotes, i.e. `"%stringConstant%"`, I'm not entirely sure whether it's clearer with or without.